rxbinding/src/main/java/com/jakewharton/rxbinding2/InitialValueObservable.java

```diff
@@ -0,0 +1,37 @@
+package com.jakewharton.rxbinding2;
+
+import android.support.annotation.MainThread;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+
+import static com.jakewharton.rxbinding2.internal.Preconditions.checkMainThread;
+
+public abstract class InitialValueObservable<T> extends Observable<T> {
+  @Override protected final void subscribeActual(Observer<? super T> observer) {
+    if (checkMainThread(observer)) {
```

Comment:
Main thread checking seems redundant here. Shouldn't the class that extends `InitialValueObservable` be able to do the check itself? (Same with on `Skipped`.)